xapian-core/include/xapian/weight.h

```diff
@@ -463,9 +463,58 @@ class XAPIAN_VISIBILITY_DEFAULT BoolWeight : public Weight {
 
 /// Xapian::Weight subclass implementing the tf-idf weighting scheme.
 class XAPIAN_VISIBILITY_DEFAULT TfIdfWeight : public Weight {
-    /* Three character string indicating the normalizations for tf(wdf), idf and
-       tfidf weight. */
-    std::string normalizations;
+    /* These three enum classes should use integers only up to 255. Otherwise
+       during serialise(), the value may get truncated by static_cast<> and
+       the desired result will not be achieved. */
+  public:
+    /** Wdf normalizations. */
+    enum class WDF_NORM: int {
```

Comment:
While constants such as the values of an `enum` are conventionally all-caps, the `enum` type itself usually isn't.  We don't have any other `enum class` in the API so far but we do have `typedef`-ed `enum` - e.g.

```
    typedef enum {
	STEM_NONE, STEM_SOME, STEM_ALL, STEM_ALL_Z, STEM_SOME_FULL_POS
    } stem_strategy;
```

So I think this type should be `wdf_norm` or similar for consistency.

(We only don't use `enum class` yet because it was new in C++11, and for 1.4.x we require a C++11 compiler to build Xapian, but aim to support use of a non-C++11 compiler for building applications using Xapian so we avoid C++11 features in the API headers - for git master we now require a C++11 compiler for building applications too, so it's OK to start using it.)

There should probably be a space before the `:` too (there isn't a precedent since we don't have an `enum class` yet, but it's `class Foo : public Base` and this seems equivalent to that situation.)

Reply:
Updated